README.txt

File Picker is a tool to help you upload, browse and add links to files in a Drupal node. 
It is derived from the Image Picker module by the same author.


CHANGELOG.txt
commit f41df30
Author: hutch <hutch@arwystli.net>
Date:   Tue Mar 1 16:37:53 2016 +0000

    added LICENSE.txt

LICENSE.txt

commit 4ef3d0a
Author: hutch <hutch@midwales.net>
Date:   Thu Feb 27 16:54:07 2014 +0000

    fixed delete bug #2206769

CHANGELOG.txt
filepicker.functions.inc

commit da62951
Author: hutch <hutch@midwales.net>
Date:   Thu Feb 27 16:53:07 2014 +0000

    better regex

filepicker.upload.inc

commit 821f872
Author: hutch <hutch@midwales.net>
Date:   Fri Jul 5 14:46:24 2013 +0100

    mad name protection #2035537

CHANGELOG.txt
filepicker.upload.inc

commit 55f5ee1
Author: hutch <hutch@midwales.net>
Date:   Tue Jun 18 23:15:45 2013 +0100

    fixed hook clash see #2022629

CHANGELOG.txt
filepicker.admin.inc
filepicker.module
filepicker.user.inc

commit c34a7a6
Author: hutch <hutch@midwales.net>
Date:   Thu Feb 14 10:20:14 2013 +0000

    fixed bug in _filepicker_edit_file()

CHANGELOG.txt
filepicker.edit.inc

commit 1cfdccc
Author: hutch <hutch@midwales.net>
Date:   Wed Nov 7 19:23:15 2012 +0000

    moved all theme functions to filepicker.module

CHANGELOG.txt
filepicker.module
filepicker.theme.inc

commit bff6c5b
Author: hutch <hutch@midwales.net>
Date:   Sat Oct 27 17:35:48 2012 +0100

    fixed bug when public is not enabled

CHANGELOG.txt
filepicker.group.inc

commit a0dcc5a
Author: hutch <hutch@midwales.net>
Date:   Tue Aug 14 09:47:26 2012 +0100

    fixed hook clash, #1729498

CHANGELOG.txt
filepicker.admin.inc
filepicker.functions.inc
filepicker.user.inc

commit 54b07c8
Author: hutch <hutch@midwales.net>
Date:   Tue Jul 3 23:51:44 2012 +0100

    avoid render strict warnings

filepicker.admin.inc
filepicker.edit.inc
filepicker.functions.inc
filepicker.group.inc
filepicker.user.inc

commit 437faed
Author: hutch <hutch@midwales.net>
Date:   Sun Jun 17 08:15:52 2012 +0100

    removed file_directory_path()

CHANGELOG.txt
filepicker.functions.inc

commit 35bc956
Author: hutch <hutch@midwales.net>
Date:   Tue Apr 10 21:46:13 2012 +0100

    improved Wysiwyg API support

CHANGELOG.txt
filepicker_iframe.js

commit 5cdbd58
Author: hutch <hutch@midwales.net>
Date:   Sat Dec 31 10:54:46 2011 +0000

    added hook_enable

filepicker.functions.inc
filepicker.install

commit 478d091
Author: hutch <hutch@midwales.net>
Date:   Thu Sep 8 10:03:55 2011 +0100

    fixed comments form

CHANGELOG.txt
filepicker.module

commit ac849dd
Author: hutch <hutch@midwales.net>
Date:   Sat Jul 30 08:53:31 2011 +0100

    added some notice avoidance

filepicker.upload.inc

commit e563267
Author: hutch <hutch@midwales.net>
Date:   Thu Jul 21 11:02:49 2011 +0100

    changed function names containing '_process'

CHANGELOG.txt
filepicker.edit.inc

commit 87f771e
Author: hutch <hutch@midwales.net>
Date:   Thu Jun 23 09:30:06 2011 +0100

    improved admin theme feature

filepicker.module

commit 075c3fd
Author: hutch <hutch@midwales.net>
Date:   Thu Jun 23 00:07:01 2011 +0100

    added admin theme option

CHANGELOG.txt
filepicker.admin.inc
filepicker.form-elements.inc
filepicker.functions.inc
filepicker.group.inc
filepicker.module
filepicker.theme.inc
filepicker.upload.inc

commit 44c601a
Author: hutch <hutch@midwales.net>
Date:   Sat Jun 11 15:40:23 2011 +0100

    fixed currentgroup bug #1185406

CHANGELOG.txt
filepicker.functions.inc
filepicker.group.inc

commit fee830b
Author: hutch <hutch@midwales.net>
Date:   Fri Jun 3 08:18:03 2011 +0100

    added reallocate feature

CHANGELOG.txt
filepicker.admin.inc
filepicker.functions.inc
filepicker.module
filepicker.theme.inc

commit 8fc4393
Author: hutch <hutch@midwales.net>
Date:   Sat Apr 2 10:15:14 2011 +0100

    fixed doubleslash bug

CHANGELOG.txt
filepicker.functions.inc
filepicker.import.inc
filepicker.upload.inc

commit ab25aad
Author: hutch <hutch@midwales.net>
Date:   Thu Mar 31 11:39:12 2011 +0100

    cleanup for new release

CHANGELOG.txt
filepicker.admin.inc
filepicker.edit.inc

commit eaca44a
Author: hutch <hutch@midwales.net>
Date:   Mon Mar 28 11:40:08 2011 +0100

    fixed iframe tinymce, fixed public roles bug

filepicker.functions.inc
filepicker.theme.inc

commit a135c09
Author: hutch <hutch@midwales.net>
Date:   Fri Mar 18 09:52:16 2011 +0000

    fixed bug in iframe multitask

CHANGELOG.txt
filepicker.functions.inc
filepicker.module

commit 8588713
Author: hutch <hutch@midwales.net>
Date:   Thu Mar 17 23:40:41 2011 +0000

    bug in admin

filepicker.admin.inc

commit 540b7ff
Author: hutch <hutch@midwales.net>
Date:   Thu Mar 17 09:29:19 2011 +0000

    fixed bug in iframe js, added multiple file upload, fixed postgres bug #1083064

CHANGELOG.txt
filepicker.admin.inc
filepicker.css
filepicker.edit.inc
filepicker.theme.inc
filepicker.upload.inc
filepicker.user.inc
filepicker_iframe.js
filepicker_upload_link.js

commit 2f99d6d
Author: hutch <hutch@midwales.net>
Date:   Sun Mar 6 12:19:50 2011 +0000

    moved some functions back to filepicker.module

CHANGELOG.txt
filepicker.functions.inc
filepicker.install
filepicker.module

commit 006b0bd
Author: hutch <hutch@midwales.net>
Date:   Sat Mar 5 12:26:39 2011 +0000

    added private download support

CHANGELOG.txt
filepicker.functions.inc
filepicker.import.inc
filepicker.install
filepicker.module

commit 737c109
Author: hutch <hutch@midwales.net>
Date:   Sun Feb 27 15:29:55 2011 +0000

    removed cvs keyword

CHANGELOG.txt
filepicker.admin.inc
filepicker.css
filepicker.edit.inc
filepicker.form-elements.inc
filepicker.functions.inc
filepicker.group.inc
filepicker.import.inc
filepicker.install
filepicker.module
filepicker.theme.inc
filepicker.tpl.php
filepicker.upload.inc
filepicker.user.inc
filepicker_iframe.js

commit 587c055
Author: hutch <hutch@midwales.net>
Date:   Sun Feb 27 00:03:29 2011 +0000

    moved filepicker_get_colorbox_perms() to filepicker.module

filepicker.functions.inc
filepicker.module

commit 157abb5
Author: The Great Git Migration <tggm@no-reply.drupal.org>
Date:   Fri Feb 25 02:14:45 2011 +0000

    Removing translation directories


filepicker_iframe.js

/**
 * @file
 * Provides javascript for insertion of html from iframe to the body in the parent.
 */

// use jquery browser detection
(function ($) {
  filepicker_browser_detect = function() {
    if ($.browser.msie) {
      return 'msie';
    }
    else if ($.browser.safari) {
      return 'safari';
    }
    else if ($.browser.opera) {
      return 'opera';
    }
    else if ($.browser.mozilla) {
      return 'mozilla';
    }
    else {
      return 'unknown';
    }
  }
})(jQuery);

// collects settings, builds HTML string
function filepickerInsert(button) {
  // Get the form element
  var filepForm = document.getElementById('filepicker-file-form');
  if (filepForm) {
    var filepFilePath = filepForm.file_path.value;
    var filepFileTitle = filepForm.file_title.value;
    var filepFileName = filepForm.file_name.value;
    var filepFileNewWindow = Drupal.settings.filepicker_iframe.file_new_window;
    var isFCKeditor = Drupal.settings.filepicker_iframe.isFCKeditor;
    var isWysiwyg = Drupal.settings.filepicker_iframe.isWysiwyg;
    var colorbox_iframe = Drupal.settings.filepicker_iframe.colorbox_iframe;
    var node_editbody = Drupal.settings.filepicker_iframe.node_editbody;
    //var i;
    var filepInsertion;
    // build a link
    if (filepFilePath) {
      filepInsertion = '<a href="' + filepFilePath + '"' + (filepFileNewWindow ? ' target="_blank"' : '') + '>' + (filepFileTitle ? filepFileTitle : filepFileName) + '</a>';
    }

    // Get the parent window of filepicker iframe
    var win = window.opener ? window.opener : window.dialogArguments;
    if (!win) {
      if (window.parent) {
        win = window.parent;
      }
      else {
        win = top;
      }
    }

    // track down a wysiwyg editor
    var jobdone = false;
    var inst = false;
    if (win.oFCK_1) {
      inst = win.oFCK_1.InstanceName;
    }
    else if (win.oFCKeditor) {
      inst = win.oFCKeditor.InstanceName;
    }
    else if (isWysiwyg == 'yes' && win.Drupal.wysiwyg) {
      //inst = 'edit-body';
      inst = win.Drupal.wysiwyg.activeId;
    }

    if (inst) {
      if (win.FCKeditorAPI) {
        if (win.FCKeditorAPI.GetInstance(inst)) {
          win.FCKeditorAPI.GetInstance(inst).InsertHtml(filepInsertion);
          jobdone = true;
        }
      }
      // ckeditor 3.?
      if (win.CKEDITOR) {
        if (win.CKEDITOR.instances[inst]) {
          win.CKEDITOR.instances[inst].insertHtml(filepInsertion);
          jobdone = true;
        }
      }
      // tinyMCE v3
      if (win.tinyMCE) {
        if (win.tinyMCE.execInstanceCommand(inst, 'mceInsertContent', false, filepInsertion)){
          jobdone = true;
        }
      }

      // Wysiwyg API
      if (! jobdone && typeof(win.Drupal.wysiwyg.instances[inst].editor) !== "undefined") {
        if (win.Drupal.wysiwyg.instances[inst].editor !== "none") {
          if (typeof(win.Drupal.wysiwyg.instances[inst].insert) !== "undefined") {
            win.Drupal.wysiwyg.instances[inst].insert(filepInsertion);
            jobdone = true;
          }
        }
      }

    }
    // older ckeditor
    if (! jobdone && win.Drupal.ckeditorInstance && win.Drupal.ckeditorInsertHtml) {
      if (win.Drupal.ckeditorInsertHtml(filepInsertion)) {
        jobdone = true;
      }
      //else
      //  return;
    }

    //var isTinyMCE = win.document.getElementById('mce_editor_0'); // buggy
    //var isTinyMCE = win.tinyMCE; // Will be undefined if tinyMCE isn't loaded. This isn't a sure-proof way of knowing if tinyMCE is loaded into a field, but it works.
    // tinyMCE v2
    if (! jobdone && win.tinyMCE && win.tinyMCE.execCommand('mceInsertContent', false, filepInsertion)) {
      jobdone = true;
    }

    if (! jobdone) {
      var nodeBody = win.document.getElementById(node_editbody);
      var commentBody = win.document.getElementById('edit-comment-value');
      var blockBody = win.document.getElementById('edit-body-value');
      if (nodeBody) {
        filepicker_insertAtCursor(nodeBody, filepInsertion);
      }
      else if (commentBody) {
        filepicker_insertAtCursor(commentBody, filepInsertion);
      }
      else if (blockBody) {
        filepicker_insertAtCursor(blockBody, filepInsertion);
      }
    }
    if (! colorbox_iframe) {
      win.location.hash = 'body_hash';
    }
  }
}

// Copy pasted from internet but modified to detect browser
function filepicker_insertAtCursor(myField, myValue) {
  browser = filepicker_browser_detect();
  if (browser == 'msie') {
    if (document.selection) {
      myField.focus();
      //in effect we are creating a text range with zero
      //length at the cursor location and replacing it
      //with myValue
      sel = document.selection.createRange();
      sel.text = myValue;
    }
  }
  else if (browser == 'opera' || browser == 'mozilla' || browser == 'safari' ) {
    if (myField.selectionStart || myField.selectionStart == '0') {
      //Here we get the start and end points of the
      //selection. Then we create substrings up to the
      //start of the selection and from the end point
      //of the selection to the end of the field value.
      //Then we concatenate the first substring, myValue,
      //and the second substring to get the new value.
      var startPos = myField.selectionStart;
      var endPos = myField.selectionEnd;
      myField.value = myField.value.substring(0, startPos)+ myValue + myField.value.substring(endPos, myField.value.length);
    }
  }
  else {
    myField.value += myValue;
  }
}



filepicker_upload.js
// $Id$

/**
 * @file
 * Javascript functions for progress bar
 *
*/

(function($) {

  Drupal.behaviors.filepicker_upload = {
    attach: function(context) {

      var name = Drupal.settings.filepicker_upload_progress.name;
      var callback = Drupal.settings.filepicker_upload_progress.callback;
      var interval = Drupal.settings.filepicker_upload_progress.interval;
      var delay = Drupal.settings.filepicker_upload_progress.delay;
      var initmessage = Drupal.settings.filepicker_upload_progress.initmessage;
      var cancelmessage = Drupal.settings.filepicker_upload_progress.cancelmessage;

      // only once
      if ($('#filepicker-sending.processed').size()) {
        return;
      }

      $('#filepicker-upload-form', context).submit(function() {
        $("#filepicker-sending").addClass('progress');
        $("#filepicker-sending").attr('aria-live', 'polite');
        $("#filepicker-sending").addClass('processed');
        $("#filepicker-sending").html('<div class="message">' + initmessage + '</div>'+
            '<div class="bar"><div class="filled"></div></div>'+
            '<div class="percentage"></div>'+
            '<a id="filepicker_upload_progress_cancel_link" href="#">' + cancelmessage + '</a>'+
            '</div>');
        Drupal.filepicker_upload_progress_hide_timeout(delay);
        // are we using PECL uploadprogress
        if (name) {
          progress_key = $("input:hidden[name=" + name + "]").val();
          var i = setInterval(function() {
            $.getJSON(callback + '?key=' + progress_key, function(data) {
              if (data == null) {
                clearInterval(i);
                return;
              }
              $("#filepicker-sending div.message").html(data.message);
              var percentage = data.percentage;
              if (percentage >= 0 && percentage <= 100) {
                $('div.filled').css('width', percentage +'%');
                $('div.percentage').html(percentage +'%');
              }
            });
          }, interval*1000);
        }
      });

      // Hide the form and show the busy div
      Drupal.filepicker_upload_progress_hide = function() {
        $('#filepicker-upload-form').hide();
        $("#filepicker-sending").show();
        $("#filepicker_upload_progress_cancel_link").click(Drupal.filepicker_upload_progress_cancel);
      }
      // set delay
      Drupal.filepicker_upload_progress_hide_timeout = function(delay) {
        setTimeout(Drupal.filepicker_upload_progress_hide, delay*1000);
      }
      // cancel the run
      Drupal.filepicker_upload_progress_cancel = function() {
        $('#filepicker-upload-form').show();
        $("#filepicker-sending").hide();

        // "reload" the form
        window.location = window.location;
      }
    }
  };

})(jQuery);


filepicker.js
// Original Id: upload_progress.js,v 1.1 2007/11/17 06:47:47 pfournier Exp
/**
 * @file
 * Javascript functions for progress bar status on node creation forms
 *
 * @author Patrick Fournier <patrick at patrickfournier dot com>
 * @author Fabio Varesano <fvaresano at yahoo dot it>
 * Adapted by Bob Hutchinson for filepicker upload form
*/

( function ($) {

/**
 * Hide the node form and show the busy div
*/
  Drupal.filepicker_upload_progress_hide_timeout = function() {
    var delay = Drupal.settings.filepicker_upload_progress.delay;
    setTimeout('filepicker_upload_progress_hide', delay*1000);
  }

  function filepicker_upload_progress_hide() {
    $('#filepicker-upload-form').hide();
    $("#sending").show();
    $("#filepicker_upload_progress_cancel_link").click(Drupal.filepicker_upload_progress_show);
  }

  Drupal.filepicker_upload_progress_show = function() {
    $('#filepicker-upload-form').show();
    $("#sending").hide();

    // "reload" the form
    window.location = window.location;
  }

  Drupal.behaviors.filepicker = function() {
    // Attaches the upload behaviour to the filepicker upload form.
    $('#filepicker-upload-form').submit(Drupal.filepicker_upload_progress_hide_timeout);

    // form enhancement
    if ($("#edit-filepicker-quota-byrole").attr('checked')) {
      $("#wrap_filepicker_quota_role").show();
    }
    else {
      $("#wrap_filepicker_quota_role").hide();
    }
    $("#edit-filepicker-quota-byrole").change(function() {
      if ($("#edit-filepicker-quota-byrole").attr('checked')) {
        $("#wrap_filepicker_quota_role").show();
      }
      else {
        $("#wrap_filepicker_quota_role").hide();
      }
    });

    if ($("#edit-filepicker-import-enabled").attr('checked')) {
      $("#wrap-filepicker-import").show();
    }
    else {
      $("#wrap-filepicker-import").hide();
    }
    $("#edit-filepicker-import-enabled").change(function() {
      if ($("#edit-filepicker-import-enabled").attr('checked')) {
        $("#wrap-filepicker-import").show();
      }
      else {
        $("#wrap-filepicker-import").hide();
      }
    });


    if ($("#edit-filepicker-upload-progress-enabled").attr('checked')) {
      $("#wrap-filepicker-upload-progress").show();
    }
    else {
      $("#wrap-filepicker-upload-progress").hide();
    }
    $("#edit-filepicker-upload-progress-enabled").change(function() {
      if ($("#edit-filepicker-upload-progress-enabled").attr('checked')) {
        $("#wrap-filepicker-upload-progress").show();
      }
      else {
        $("#wrap-filepicker-upload-progress").hide();
      }
    });

    if ($("#edit-group-public").attr('checked')) {
      $("#wrap-group-public-roles").show();
    }
    else {
      $("#wrap-group-public-roles").hide();
    }
    $("#edit-group-public").change(function() {
      if ($("#edit-group-public").attr('checked')) {
        $("#wrap-group-public-roles").show();
      }
      else {
        $("#wrap-group-public-roles").hide();
      }
    });
  };

})(jQuery);
